xlator/options.go

```diff
@@ -65,3 +71,176 @@ type Option struct {
+		err = ValidatePath(o, val)
+	case OptionTypePercent:
+		err = ValidatePercent(o, val)
+	case OptionTypePercentOrSizet:
+		err = ValidatePercentOrSize(o, val)
+	case OptionTypePriorityList:
+		err = ValidatePriorityOrSize(o, val)
+	case OptionTypeSizeList:
+		err = ValidateSizeList(o, val)
+	case OptionTypeSizet:
+		err = ValidateSizet(o, val)
+	case OptionTypeStr:
+		err = ValidateStr(o, val)
+	case OptionTypeTime:
+		err = ValidateTime(o, val)
+	case OptionTypeXlator:
+		err = ValidateXlator(o, val)
+	default:
+		err = ValidateOption(o, val)
+	}
+	return err
+}
+
+// ValidateBool validates if the option is of type boolean
+func ValidateBool(o *Option, val string) error {
+	if val == "" {
+		err := errors.New("No argument passed")
+		return err
+	} else if strings.EqualFold(val, "1") == true ||
+		strings.EqualFold(val, "on") == true ||
+		strings.EqualFold(val, "yes") == true ||
+		strings.EqualFold(val, "true") == true ||
+		strings.EqualFold(val, "enable") == true ||
+		strings.EqualFold(val, "0") == true ||
+		strings.EqualFold(val, "off") == true ||
+		strings.EqualFold(val, "no") == true ||
+		strings.EqualFold(val, "false") == true ||
+		strings.EqualFold(val, "disable") == true {
+		return nil
+	} else {
+		err := errors.New("Invalid argument")
+		return err
+	}
+}
+
+// ValidateClientAuthAddr validates mount auth address
+func ValidateClientAuthAddr(o *Option, val string) error {
+	err := errors.New("Argument invalid")
+	return err
+}
+
+//ValidateDouble validates if the option is of type double
+func ValidateDouble(o *Option, val string) error {
+	err := errors.New("Argument invalid")
+	return err
+}
+
+// ValidateInt validates if the option is of type Int
+func ValidateInt(o *Option, val string) error {
+	err := errors.New("Argument invalid")
+	return err
+}
+
+// ValidateInternetAddress validates the Internet Address
+func ValidateInternetAddress(o *Option, val string) error {
+	err := errors.New("Argument invalid")
+	return err
+}
+
+// ValidateInternetAddressList validates the Internet Address List
+func ValidateInternetAddressList(o *Option, val string) error {
+	err := errors.New("Argument invalid")
+	return err
+}
+
+// ValidatePath validates if the option is a valid path
+func ValidatePath(o *Option, val string) error {
+	var err error
+	if strings.Contains(val, "../") == true {
+		err = errors.New("invalid path given")
+	} else if val[0] != '/' {
+		err = errors.New("option is not an absolute path name")
+	}
+	return err
+}
+
+// ValidatePercent validates if the option is in correct percent format
+func ValidatePercent(o *Option, val string) error {
+	var percent float64
+	var err error
+	l := len(val)
+	if l == 1 && strings.ContainsRune(val, '%') {
+		err = errors.New("Invalid percent format")
+	} else if l > 1 && strings.ContainsRune(val, '%') {
+		if val[0] == '%' {
+			err = errors.New("Invalid percent format")
+		}
+		percent, err = strconv.ParseFloat(val[:(l-1)], 64)
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
This should accept percents with and without the percent symbol. The current validation fails if a the symbol isn't present.

Reply:
Ack, will do.